src/Uno.UI/DataBinding/BindingPropertyHelper.FastConvert.cs

```diff
@@ -364,54 +364,185 @@ private static bool FastStringToInputScope(Type outputType, string input, ref ob
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.ChineseHalfWidth;
+			}
+			else if (trimmed.Equals("54", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("chinesefullwidth", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.ChineseFullWidth;
+			}
+			else if (trimmed.Equals("55", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("nativescript", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.NativeScript;
+			}
+			else if (trimmed.Equals("57", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("text", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.Text;
+			}
+			else if (trimmed.Equals("58", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("chat", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.Chat;
+			}
+			else if (trimmed.Equals("59", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("nameorphonenumber", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.NameOrPhoneNumber;
+			}
+			else if (trimmed.Equals("60", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("emailnameoraddress", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.EmailNameOrAddress;
+			}
+			else if (trimmed.Equals("62", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("maps", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.Maps;
+			}
+			else if (trimmed.Equals("63", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("numericpassword", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.NumericPassword;
+			}
+			else if (trimmed.Equals("64", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("numericpin", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.NumericPin;
+			}
+			else if (trimmed.Equals("65", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("alphanumericpin", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.AlphanumericPin;
+			}
+			else if (trimmed.Equals("67", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("formulanumber", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.FormulaNumber;
+			}
+			else if (trimmed.Equals("68", StringComparison.OrdinalIgnoreCase) || trimmed.Equals("chatwithoutemoji", StringComparison.OrdinalIgnoreCase))
+			{
+				nameValue = Windows.UI.Xaml.Input.InputScopeNameValue.ChatWithoutEmoji;
+			}
+			else
+			{
+				throw new InvalidOperationException($"Failed to create a '{outputType.FullName}' from the text '{input}'.");
+			}
+
```

Comment:
Do you have a benchmark that shows the gains ? like the old version, this version and a mix of using `Span` and `switch` ?

Nice things about `switch` is that 

* it's much easier to read the code

* the compiler (current and future) can optimize the code - having a benchmark allow anyone to compare numbers again to see if the extra code is still worth it.

Reply:
I guess that it may only reduce allocations, and only if tolowerinvariant reallocates even if the string is the same.

Reply:
Yes, it should reduce allocations (and `ToLowerInvariant` should not allocate anything on a lowercased string).

However a smart compiler [1] should be able to create a collision-free, case-insensitive hash code (and without any allocation) and use this for a faster switch. Right now the last `if ` conditions will be much slower than the first ones. That would not matter for smaller sets of values but this one is rather large.

[1] Roslyn might not be there yet and/or we could do this manually but probably not worth the dev time ;-)

Anyway the `switch` code is more optimization friendly (today or in the future) than a series of `if ... else if ...` and having a _canary_ benchmark could alert us if the situation change :-)

Reply:
Okay, let me close this out.